src/header/header.scss

```diff
@@ -0,0 +1,227 @@
+      flex-shrink: 0;
+      margin: 0 $iui-xs;
+    }
+  }
+
+  > .iui-center {
+    justify-content: center;
+    flex: 1 1 auto;
+  }
+
+  > .iui-right {
+    flex: 2 1 33%;
+    justify-content: flex-end;
+
+    .iui-user-icon {
+      transition: width $iui-speed-fast ease-out, height $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out;
+        
+      * {
+        transition: width $iui-speed-fast ease-out, height $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out, line-height $iui-speed-fast ease-out, font-weight $iui-speed-fast ease-out;
+      }
+    }
+  }
+
+  .iui-button {
+    height: 100%;
+    border-radius: 0;
+  }
+
+  .iui-divider {
+    width: 1px;
+    height: $iui-baseline * 2;
+    @include themed {
+      background-color: t(iui-color-background-4);
+    }
+  }
+  
+  &.iui-slim,
+  &.iui-slim-on-hover-demo:hover {
+    height: round($iui-baseline * 2.5);
+
+    > .iui-left {
+      > .iui-application {
+        > .iui-label {
+          font-size: $iui-font-size;
+          margin-right: $iui-l;
+        }
+
+        > .iui-brand-logo > .iui-icon {
+          width: $iui-icons-default;
+          height: $iui-icons-default;
+        }
+      }
+
+      > .iui-header-button {
+        > .iui-thumbnail-area > .iui-thumbnail {
+          width: $iui-icons-default;
+          height: $iui-icons-default;
+        }
+
+        &.iui-detailed > .iui-label {
+          padding-top: $iui-baseline * 2;
+
+          > .iui-id {
+            opacity: 0;
+          }
+        }
+      }
+    }
+    
+    > .iui-right {
+      .iui-user-icon {
+        @include iui-user-icon;
+        @include iui-user-icon-size($size: $iui-icons-large, $font-size: $iui-font-size-small);
+      }
+    }
+  }
+}
+
+@mixin iui-header-buttons {
+  > .iui-application,
+  > .iui-header-button {
+    overflow: hidden;
+    transition: height $iui-speed-fast ease-out;
+
+    > .iui-label {
+      overflow: hidden;
+      text-overflow: ellipsis;
+      transition: font-size $iui-speed-fast ease-out;
+
+      > * {
+        width: 100%;
+        overflow: hidden;
+        text-overflow: ellipsis;
+        text-align: left;
+      }
+    }
+
+    > .iui-brand-logo,
+    > .iui-thumbnail-area {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      height: 100%;
+
+      > .iui-icon,
+      > .iui-thumbnail {
+        width: $iui-icons-large;
+        height: $iui-icons-large;
+        transition: width $iui-speed-fast ease-out, height $iui-speed-fast ease-out;
+      }
+    }
+  }
+
+  > .iui-application {
+    padding: 0;
+
+    > .iui-brand-logo {
+      width: $iui-l * 3 - 1;
+      flex-shrink: 0;
+    }
+
+    > .iui-label {
+      margin-right: $iui-sm;
+      transition: margin-right $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out;
+    }
+  }
+
+  > .iui-header-button {
+    > .iui-thumbnail-area {
+      width: $iui-l;
+      margin-right: $iui-sm;
+
+      > .iui-thumbnail {
```

Comment:
Wouldnt this be handled by leftIcon prop in button? So we would not need these additional handlers...